src/main/java/com/sforce/ws/bind/XmlObject.java

```diff
@@ -302,6 +330,42 @@ protected void loadAfterStartTag(XmlInputStream in, TypeMapper typeMapper) throw
         if (textFound) value = typeMapper.deserialize(text.toString(), xmlType);
     }
 
+    private XmlObject extractChildElement(XmlInputStream in, TypeMapper typeMapper) throws ConnectionException {
+        XmlObject child;
+        QName xsiType = typeMapper.getXsiType(in);
+        if (xsiType == null) {
+            child = new XmlObject();
+        } else if (xsiType.getLocalPart().equals("QueryResult") && !in.getName().equals("QueryResult")) {
+            // In older API versions, some objects had their xsitype erroneously set to "QueryResult"
+            // even when they were not QueryResult objects. Since we have no information about their type
+            // we will default back to XmlObject.
+            child = new XmlObject();
+        } else {
+            Class<?> childClass = typeMapper.getJavaType(xsiType);
+            if (childClass == null || !XMLizable.class.isAssignableFrom(childClass)) {
+                child = new XmlObject();
+            }
+            else {
+                XMLizable xmlizable;
+                try {
+                    xmlizable = (XMLizable)childClass.newInstance();
+                    if (xmlizable instanceof XmlObject) {
+                        child = (XmlObject) xmlizable;
+                    } else {
+                        child = new XmlObjectWrapper(xmlizable);
+                    }
+                } catch (InstantiationException e) {
+                    throw new ConnectionException("Failed to create object", e);
+                } catch (IllegalAccessException e) {
+                    throw new ConnectionException("Failed to create object", e);
```

Comment:
Yep, I did have that there before, but WSC uses JDK 1.6. I didn't want to mix in upgrading it to 1.7 here.